pkg/cmd/util/clientcmd/factory_client_access.go

```diff
@@ -51,7 +63,22 @@ type ring0Factory struct {
 type ClientAccessFactory interface {
 	kcmdutil.ClientAccessFactory
 
+	// TODO: this should be removed when we finally get rid of pkg/client
 	Clients() (*client.Client, kclientset.Interface, error)
+	// use below clients instead whenever needed:
+	OpenShiftAppsClient() (appsclient.Interface, error)
+	OpenShiftAuthorizationClient() (authorizationclient.Interface, error)
```

Comment:
@deads2k do we need this client or we want upstream?

Reply:
> @deads2k do we need this client or we want upstream?

This client points to our resources and not all of them will move upstream.